library/src/CMakeLists.txt

```diff
@@ -24,7 +24,91 @@
+      src/amd_detail/precond/hipsparse_csrilu0.cpp
+      src/amd_detail/precond/hipsparse_gpsv_interleaved_batch.cpp
+      src/amd_detail/precond/hipsparse_gtsv_interleaved_batch.cpp
+      src/amd_detail/precond/hipsparse_gtsv_nopivot.cpp
+      src/amd_detail/precond/hipsparse_gtsv_strided_batch.cpp
+      src/amd_detail/precond/hipsparse_gtsv.cpp
+      # Extra
+      src/amd_detail/extra/hipsparse_csrgeam.cpp
+      src/amd_detail/extra/hipsparse_csrgemm.cpp
+      # Generic
+      src/amd_detail/generic/hipsparse_axpby.cpp
+      src/amd_detail/generic/hipsparse_gather.cpp
+      src/amd_detail/generic/hipsparse_scatter.cpp
+      src/amd_detail/generic/hipsparse_rot.cpp
+      src/amd_detail/generic/hipsparse_spvv.cpp
+      src/amd_detail/generic/hipsparse_spmv.cpp
+      src/amd_detail/generic/hipsparse_spsv.cpp
+      src/amd_detail/generic/hipsparse_spmm.cpp
+      src/amd_detail/generic/hipsparse_spsm.cpp
+      src/amd_detail/generic/hipsparse_sddmm.cpp
+      src/amd_detail/generic/hipsparse_spgemm.cpp
+      src/amd_detail/generic/hipsparse_spgemm_reuse.cpp
+      src/amd_detail/generic/hipsparse_sparse2dense.cpp
+      src/amd_detail/generic/hipsparse_dense2sparse.cpp
+      # Conversion
+      src/amd_detail/conversion/hipsparse_bsr2csr.cpp
+      src/amd_detail/conversion/hipsparse_coo2csr.cpp
+      src/amd_detail/conversion/hipsparse_coosort.cpp
+      src/amd_detail/conversion/hipsparse_create_identity_permutation.cpp
+      src/amd_detail/conversion/hipsparse_csc2dense.cpp
+      src/amd_detail/conversion/hipsparse_cscsort.cpp
+      src/amd_detail/conversion/hipsparse_csr2bsr.cpp
+      src/amd_detail/conversion/hipsparse_csr2coo.cpp
+      src/amd_detail/conversion/hipsparse_csr2csc.cpp
+      src/amd_detail/conversion/hipsparse_csr2csr_compress.cpp
+      src/amd_detail/conversion/hipsparse_csr2csru.cpp
+      src/amd_detail/conversion/hipsparse_csr2dense.cpp
+      src/amd_detail/conversion/hipsparse_csr2gebsr.cpp
+      src/amd_detail/conversion/hipsparse_csr2hyb.cpp
+      src/amd_detail/conversion/hipsparse_csrsort.cpp
+      src/amd_detail/conversion/hipsparse_csru2csr.cpp
+      src/amd_detail/conversion/hipsparse_dense2csc.cpp
+      src/amd_detail/conversion/hipsparse_dense2csr.cpp
+      src/amd_detail/conversion/hipsparse_gebsr2csr.cpp
+      src/amd_detail/conversion/hipsparse_gebsr2gebsc.cpp
+      src/amd_detail/conversion/hipsparse_gebsr2gebsr.cpp
+      src/amd_detail/conversion/hipsparse_hyb2csr.cpp
+      src/amd_detail/conversion/hipsparse_nnz_compress.cpp
+      src/amd_detail/conversion/hipsparse_nnz.cpp
+      src/amd_detail/conversion/hipsparse_prune_csr2csr_by_percentage.cpp
+      src/amd_detail/conversion/hipsparse_prune_csr2csr.cpp
+      src/amd_detail/conversion/hipsparse_prune_dense2csr_by_percentage.cpp
+      src/amd_detail/conversion/hipsparse_prune_dense2csr.cpp
+      # Reorder
+      src/amd_detail/reorder/hipsparse_csrcolor.cpp
+      )
```

Comment:
I am not sure if this is on purpose to list all files explicitly but there are alternatives to include all `*.cpp` files if preferred.

Reply:
FILE(GLOB AMD_DETAIL_SOURCE *.cpp)

Reply:
btw, it may be better to have a cmake script in amd_detail and another one in nvidia_detail.
See hipGRAPH-internal.